File: B_Squares_and_Cubes.py
t = input()
t = int(t)

while t > 0:
    n = input()
    n = int(n)
    t = t-1

    count = 1
    i = 1
    while i < n:
        if((i*i) <= n and (i*i*i) <= n):
            count = count+1
        
        elif((i*i) <= n or (i*i*i) <= n):
            count = count+1
        
        i = i+1
    print(count)

File: B_Squares_and_Cubes.cpp
#include <bits/stdc++.h>
using namespace std;

int main()
{
    int t;
    cin>>t;

    while(t--){
        long long n;
        cin>>n;

        int count = 1;
        for (int i = 1; i*i <= n; i++){

            int x = i*i;
            int y = i*i*i;

            if(x == y){
                count++;
            }
            else{
                count += 2;
            }
            
        }
        cout<<count<<endl;
      
    }
    
    return 0;
}